### DISCUSSION TITLE
Finding out the current position of a pointer with respect to a control or a screen

### DISCUSSION THREAD

**Discussion by npolyak** [AvaloniaUI/Avalonia#6386] [Q&A] [answered]
I know that one can use PointerEventArgs.GetPosition(...) method, but what is we are not inside a pointer event handler. WPF has Mouse.GetPosition(...) method for those cases, what does AvaloniaUI have?

**Comment by maxkatz6** [comment]
WPF API is not multiple pointers friendly. Especially with multitouch.
But if you are very sure that it's fine, you can use this deprecated method:
```
window.PlatformImpl.MouseDevice.GetPosition(relativeTo);
```
Or better to subscribe on window's PointerMoved event and save last known position to some kind of property, if it's exactly what you need.

Related: https://github.com/AvaloniaUI/Avalonia/discussions/6201#discussioncomment-972437
And: https://github.com/AvaloniaUI/Avalonia/pull/2581#discussion_r288640943

**Reply by npolyak** [reply]
I tried MouseDevice - it is still populated in a 'push' way based on the events related to the window. What if a pointer is not above the window?

**Comment by kekekeks** [comment, marked as answer]
There are some features that we don't consider to be portable. You can, however, always use platform APIs.

Windows: https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getcursorpos
X11: https://tronche.com/gui/x/xlib/window-information/XQueryPointer.html
macOS:
either P/Invoke to replicate something like 
```
CGEventRef ourEvent = CGEventCreate(NULL);
point = CGEventGetLocation(ourEvent);
CFRelease(ourEvent);
```
or use MonoMac.NetStandard package to get NSEvent.MouseLocation

**Reply by kekekeks** [reply]
Note, that you won't be able to get the global mouse location once we implement Wayland backend. There is _no such thing as global coordinates_ with Wayland.
